corehq/ex-submodules/casexml/apps/phone/restore.py

```diff
@@ -221,18 +221,30 @@ def get_http_response(self):
 class CachedResponse(object):
 
     def __init__(self, name):
+        if name and name.startswith("restore-response-"):
+            # Name template was 'restore-response-{}.xml' before new
+            # blob metadata API was implemented. This can be removed
+            # when all old responses have expired.
+            name = "_default/" + name
         self.name = name
 
     @classmethod
-    def save_for_later(cls, fileobj, timeout):
+    def save_for_later(cls, fileobj, timeout, domain):
         """Save restore response for later
 
         :param fileobj: A file-like object.
         :param timeout: Minimum content expiration in seconds.
         :returns: A new `CachedResponse` pointing to the saved content.
         """
-        name = 'restore-response-{}.xml'.format(uuid4().hex)
-        get_blob_db().put(NoClose(fileobj), name, timeout=max(timeout // 60, 60))
+        name = 'restore-{}.xml'.format(uuid4().hex)
+        get_blob_db().put(
+            NoClose(fileobj),
+            domain=domain,
+            parent_id=domain,
```

Comment:
this will hash all these to the same SQL DB which could create a hotspot for ICDS. What about using the restore user ID?

Reply:
Good idea. 7ed7942c01c8cc4ee7a62630d5d7afc5a601c394